packages/wallet-apis/src/provider.ts

```diff
@@ -61,7 +64,7 @@ export const createEip1193ProviderFromClient = <
     try {
       switch (method) {
         case "eth_chainId": {
-          return handler<"eth_chainId">(async () => {
+          return await handler<"eth_chainId">(async () => {
```

Comment:
went down a rabbit hole with this, didn't know that `return await` is the new meta in try/catch now. ty for fixing

Reply:
yeahh..... this has bitten me a few times tbh. i only learned this a few months ago. didn't notice until i added tests & the errors were not the expected type.